### PR TITLE
Adds Basic Google Books Calling Functionality

### PR DESCRIPTION
**Pull request by LilAttacker**
This is a basic version of google books call functionality. To be expanded upon later. Much of it is currently hardcoded, but it does have some variables already added to make modification easier later.

Allows the ability to make a call to Google Books and another method to return the titles of said results. Also only makes a call when a certain method is called by the server. This is labelled within the code.

Includes other documentation for future reference in regards to API key creation within the code. That is not implemented yet.

Should be noted, I forgot to include the update to necessary packages to easily install the package "requests" so you can actually make api calls.  The command I used to install it is in a comment within the code in "resource_format".

- [ ] New Feature
- [ ] Documentation Update

No new warnings.
I tried to follow the code style, notify me if I got it wrong.
I have added some notes. I'll add more when I return.

## Additional Notes

If this isn't accepted when I get back to this, I'll update the required packages list then.

### CODE REVIEW

**Comment by LilAttacker** [comment]
Updated to finally include appropriate package requirements in poetry.

Refactored resource_format.py.
- Removed unnecessary functions.
- Removed global variable.
- Added new simpler function that returns key book attributed returned from the API.
- Now API calls refer to a .env file that will hold the API key rather than be hard coded.

Added some tests to test_api.py.

**IMPORTANT**
There seems to be an issue in the way that pytest and python import packages in that they do it differently from each other. In testing, both would work flawlessly when using their individual import methods, but would refuse to work when trying the other options. I will conduct additional research into this later, but as of now I don't have a solution and there doesn't actually seem to be a good fix but I'm hoping I'm wrong. Regardless, it still works when using appropriate import methods.

**TLDR: In certain files (Including __main__.py and indicated in there as well) you have to swap the import methods depending on if you are testing (pytest) or running (python). I made comments to indicate appropriate method.**

**Comment by ederic-oytas** [comment]
I approve this pull request just so we can get the code on the repo.

Before pulling, Daniel, could you do `git pull` and resolve the merge conflicts before this is pushed? This should also get the Github workflows for testing. Make sure to keep the current test bed in the repo.

In addition, remember our code style: use `lower_case` for functions and variables and `PascalCase` for classes. For example:
- `GoogleBooksTestCallAPI()` to `google_books_test_call_api()`
- `api_testing` to `ApiTesting` or `APITesting`.

Thank you, Daniel, for your hard work!

**Comment by LilAttacker** [comment]
It took quite a few tries (honestly too many, flake8 hates my formatting so much) but it passes the tests. I changed and updated some things, but it should still work.

There is still the weird importing issue with pytest vs python, but I changed the way the routing works to (hopefully) be more readable and it had the nice effect of making it way easier to swap the importing format. You now only need to swap the import statement in __main__.py and no where else when going from pytest to python.

File: server/tabby_server/services/resource_format.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    if bool(items):
+        # List that holds all the dicts for each book.
+        all_books = []
+
+        for i in range(len(items)):
+            # Dict to hold the individual book attributes
+            entry = {}
+
+            try:
+                entry['isbn'] = items[i]["volumeInfo"]["industryIdentifiers"]
```
Comment (David1000734) [inline server/tabby_server/services/resource_format.py:97]: I can see how this would work however it is kinda ugly. I believe there has to be some function like try_default_if_null or something that will try to get something out from that JSON and will not throw an exception if it fails. If not that, there has to be a better way to have this work.

File: server/tabby_server/__main__.py [hunk 2]
```diff
@@ -1,9 +1,39 @@
...
+# --- IF USING PYTEST VS PYTHON ---
+# You only need to modify this file! I've changed it around so only this
+# breaks pytest or python! No need to modfy additional files beyond!
 
 app = Flask(__name__)
 
 
+# @app.route('/testpage', methods=['GET'])
+# app.add_url_rule('/testpage', view_func=alternate_page.test_page)
+# @app.route('/test/make_request', methods=['GET'])
```
Comment (David1000734) [inline server/tabby_server/__main__.py:16]: This API route appears to be lazy loading where it would be difficult to specify what kind of HTTP Route we want. Can we look into a better way of implementing the separate files. For example,
We can leave the app.routes same as how we had it before and if we want to have the code in a separate file, then we can have it call a function that is external to the main. Then main would just import that external file for the function and all that.

**Review by David1000734 — APPROVED** [review]
Code looks good. It would be nice to have some optimizations like that try catch and lazy loading fixed.
